src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,18 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (strcmp(arg, "map") == 0) {
```

Comment:
@liupan1111 Make a temporary copy of the argument vector, run  ```md_config_t().parse_argv(args);``` against the temporary copy to strip out common command-line optionals, and continue the existing processing in ```rbd_nbd``` function using the (pruned) temporary copy. When you call ```do_map```, pass along the original (unmodified) argument vector so that global_init can re-process the command-line optionals.